core/src/test/java/io/grpc/ClientInterceptorsTest.java

```diff
@@ -351,6 +352,33 @@ protected void checkedStart(ClientCall.Listener<RespT> responseListener, Metadat
     ArgumentCaptor<Status> captor = ArgumentCaptor.forClass(Status.class);
     verify(listener).onClose(captor.capture(), any(Metadata.class));
     assertSame(error, captor.getValue().getCause());
+
+    // Make sure nothing bad happens after the exception.
+    ClientCall<?, ?> noop = ((CheckedForwardingClientCall<?, ?>)interceptedCall).delegate();
```

Comment:
The mocking used in this test actually seems reasonable. It is verifing interactions, and there should be exactly _zero_ with the `call`. It will be exactly as brittle as any other test that verifies that `call` was not called.

> The code above probably isn't doing what you think it is, because it didn't cover the lines that this addition is.

I ran:
```
gradle clean test --tests='*ClientInterceptorsTest.exceptionInStart' jacocoTestReport
```

And I see NOOP_CALL's request, halfClose and sendMessage methods are covered. That leaves isReady and cancel, and start isn't really expected to be called.